src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -0,0 +1,205 @@
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.VersionTools.Automation
+{
+    public class PullRequestCreator
+    {
+        private GitHubAuth _auth;
+
+        public string BranchPrefix { get; set; } = "UpdateDependencies";
+
+        public bool AlwaysCreateNewPullRequest { get; set; }
+
+        public GitHubBranch UpstreamBranch { get; }
+        public GitHubProject Origin { get; }
+
+        public string GitAuthorName { get; }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            string projectName,
+            string upstreamOwner,
+            string upstreamBranch,
+            string gitAuthorName)
+            : this(
+                auth,
+                new GitHubProject(projectName, auth.User),
+                new GitHubBranch(upstreamBranch, new GitHubProject(projectName, upstreamOwner)),
+                gitAuthorName)
+        {
+        }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            GitHubProject origin,
+            GitHubBranch upstreamBranch,
+            string gitAuthorName = null)
+        {
+            _auth = auth;
+
+            Origin = origin;
+            UpstreamBranch = upstreamBranch;
+
+            GitAuthorName = gitAuthorName ?? auth.User;
+        }
+
+        public async Task CreateAsync(
+            string commitMessage,
+            string title,
+            string description)
+        {
+            using (var client = new GitHubClient(_auth))
+            {
+                var originBranch = new GitHubBranch(
+                    $"{BranchPrefix}{DateTime.UtcNow.ToString("yyyyMMddhhmmss")}",
+                    Origin);
+
+                GitHubPullRequest pullRequestToUpdate = null;
+
+                if (!AlwaysCreateNewPullRequest)
+                {
+                    pullRequestToUpdate = await client.FirstPullRequestWithHeadOrDefaultAsync(
+                        Upstream,
+                        BranchPrefix,
+                        _auth.User);
+
+                    if (pullRequestToUpdate == null)
+                    {
+                        Trace.TraceInformation($"No existing pull request found.");
+                    }
+                    else
+                    {
+                        Trace.TraceInformation(
+                            $"Pull request already exists for {BranchPrefix} in {Upstream.Segments}. " +
+                            $"#{pullRequestToUpdate.Number}, '{pullRequestToUpdate.Title}'");
+
+                        if (CanUpdatePullRequest(client, pullRequestToUpdate))
+                        {
+                            originBranch = new GitHubBranch(
+                                pullRequestToUpdate.Head.Ref,
+                                originBranch.Project);
+                        }
+                        else
+                        {
+                            Trace.TraceWarning($"Can't update pull request #{pullRequestToUpdate.Number}.");
```

Comment:
Good points. I'll change this to add a comment with the specific reason. (Comment because I realized those send emails, but changes to the PR body don't.)